processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -54,18 +69,36 @@ func (d *Detector) Detect(_ context.Context) (resource pdata.Resource, schemaURL
 		return res, "", fmt.Errorf("failed getting OS type: %w", err)
 	}
 
-	hostname, err := d.provider.FQDN()
-	if err != nil {
-		// Fallback to OS hostname
-		d.logger.Debug("FQDN query failed, falling back to OS hostname", zap.Error(err))
-		hostname, err = d.provider.Hostname()
-		if err != nil {
-			return res, "", fmt.Errorf("failed getting OS hostname: %w", err)
+	for _, source := range d.hostnameSources {
+		getHostname := hostnameSourcesMap[source]
+		hostname, err = getHostname(d)
+		if err == nil {
+			attrs.InsertString(conventions.AttributeHostName, hostname)
+			attrs.InsertString(conventions.AttributeOSType, osType)
+
+			return res, conventions.SchemaURL, nil
+		} else {
+			errs = append(errs, err)
 		}
```

Comment:
Unnecessary else
```suggestion
		}
		errs = append(errs, err)
```

Reply:
I have added the suggested changes. Please, review the same.